### PR TITLE
Add root saturation function (issue #702)

### LINKED ISSUE #702: Add root saturation function
I liked the idea of the root saturation that came up in #614 

Seems easy to add and is easy to use in models after #632 allowing others to try!

```python

from pymc_marketing.prior import Prior
from pymc_marketing.mmm.components.saturation import SaturationTransformation

# pymc_marketing/mmm/transformations
def root_saturation(x, alpha): 
    return x ** alpha

# pymc_marketing/mmm/components/saturation.py
class RootSaturation(SaturationTransformation): 
    def function(self, x, alpha, beta): 
        return beta * root_saturation(x, alpha)

    # More thought required here for sensible priors
    default_priors = {
        "alpha": Prior("Beta", alpha=1, beta=2),
        "beta": Prior("Gamma", mu=1, sigma=1),
    }

# Added below in the dictionary for string access
```

What are your thoughts here? @louismagowan

### PR DESCRIPTION
**Pull request by iraur**
## Description
Adding the ability to saturate via the root function as discussed in issue [#702](https://github.com/pymc-labs/pymc-marketing/issues/702) in order to further increase the list of standard saturation functions available to users of pymc-marketing

## Related Issue

- [x] Closes #702 

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] MMM
- [ ] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--858.org.readthedocs.build/en/858/

### CODE REVIEW

**Comment by wd60622** [comment]
Thanks for opening up the PR @iraur 
Would you be able to install pre-commit and run the hooks for the files you've changed

**Comment by wd60622** [comment]
The plots don't seem to be showing
- https://pymc-marketing--858.org.readthedocs.build/en/858/api/generated/pymc_marketing.mmm.components.saturation.RootSaturation.html
- https://pymc-marketing--858.org.readthedocs.build/en/858/api/generated/pymc_marketing.mmm.transformers.root_saturation.html

File: pymc_marketing/mmm/transformers.py [hunk 1]
```diff
@@ -985,3 +985,52 @@ def hill_saturation(
...
+    alpha : float
+        Exponent for the root transformation. Must be non-negative.
+
+    Returns
+    -------
+    tensor
+        Transformed tensor.
+
+    References
+    ----------
```
[the remark this answers is not in the document]
Comment (iraur) [reply pymc_marketing/mmm/transformers.py:1033]: Thanks! Agree - I'll remove this section

File: pymc_marketing/mmm/transformers.py [hunk 2]
```diff
@@ -985,3 +985,52 @@ def hill_saturation(
...
+        import matplotlib.pyplot as plt
+        import numpy as np
+        import arviz as az
+        from pymc_marketing.mmm.transformers import root_saturation
+        plt.style.use('arviz-darkgrid')
+        alpha_values = [0.25, 0.5, 0.75, 1, 1.25]
+        x = np.linspace(0, 5, 100)
+        ax = plt.subplot(111)
+        for alpha in alpha_values:
+            y = root_saturation(x, alpha=alpha).eval()
```
[the remark this answers is not in the document]
Comment (iraur) [reply pymc_marketing/mmm/transformers.py]: Thanks! Agree and removed! ✅

**Comment by iraur** [comment]
> The plots don't seem to be showing
> 
> * https://pymc-marketing--858.org.readthedocs.build/en/858/api/generated/pymc_marketing.mmm.components.saturation.RootSaturation.html
> * https://pymc-marketing--858.org.readthedocs.build/en/858/api/generated/pymc_marketing.mmm.transformers.root_saturation.html

Thanks for pointing this out @wd60622 ! I've pushed a fix for that. Have just noticed the graph isn't produced for the Hill function ([see here](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.mmm.transformers.hill_saturation.html)) - shall I push a fix for that?

Was also wondering whether including log saturation might be of interest? Something like f(x; a,b) = a * log(x + b) ? What do you think about that - maybe overkill?

**Comment by wd60622** [comment]
> Thanks for pointing this out @wd60622 ! I've pushed a fix for that. Have just noticed the graph isn't produced for the Hill function ([see here](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.mmm.transformers.hill_saturation.html)) - shall I push a fix for that?


That was just addressed in another issue. I merged it into your branch so should be good to go on that end

> Was also wondering whether including log saturation might be of interest? Something like f(x; a,b) = a * log(x + b) ? What do you think about that - maybe overkill?

Let's get this merged in and we can address in separate PR

File: pymc_marketing/mmm/components/saturation.py [hunk 3]
```diff
@@ -369,6 +370,40 @@ class HillSaturation(SaturationTransformation):
...
+        plt.show()
+
+    """
+
+    def function(self, x, alpha, beta):
+        return beta * root_saturation(x, alpha)
+
+    lookup_name = "root"
+
+    # REMINDER
```
Comment (wd60622) [inline pymc_marketing/mmm/components/saturation.py:400]: We can remove

File: pymc_marketing/mmm/transformers.py [hunk 4]
```diff
@@ -988,3 +988,48 @@ def hill_saturation(
...
+    alpha: pt.TensorLike,
+) -> pt.TensorVariable:
+    r"""Root saturation transformation.
+
+    .. math::
+        f(x) = x^{\alpha}
+
+    .. plot::
+        :context: close-figs
+        import matplotlib.pyplot as plt
```
Comment (wd60622) [inline pymc_marketing/mmm/transformers.py:1004]: Might need a space here

**Review by wd60622 — CHANGES_REQUESTED** [review]
Can you add the RootSaturation to the list of saturation transformations in the tests/mmm/components/test_saturation.py file

File: pymc_marketing/mmm/components/saturation.py [hunk 5]
```diff
@@ -369,6 +370,39 @@ class HillSaturation(SaturationTransformation):
...
+    """Wrapper around Root saturation function.
+
+    For more information, see :func:`pymc_marketing.mmm.transformers.root_saturation`.
+
+    .. plot::
+        :context: close-figs
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+        from pymc_marketing.mmm import RootSaturation
```
Comment (wd60622) [inline pymc_marketing/mmm/components/saturation.py:383]: This needs to be added to the pymc_marketing/mmm/__init__.py file for this to work

**Review by wd60622 — CHANGES_REQUESTED** [review]
Almost there. Just need to add to mmm/init file

File: pymc_marketing/mmm/components/saturation.py (hunk 5 above)
Comment (iraur) [reply pymc_marketing/mmm/components/saturation.py:383]: Ahh thanks so much @wd60622 for hand holding me though this - really appreciate it! That's been done ✅

Comment (wd60622) [reply pymc_marketing/mmm/components/saturation.py:383]: My pleasure

**Review by wd60622 — APPROVED** [review]
Looks great! Thanks @iraur

**Pull request merged**